py/server/deephaven/dbc/__init__.py

```diff
@@ -1,5 +1,35 @@
 #
 #     Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
 #
+"""The dbc package includes the modules and functions for using external databases with Deephaven."""
 
-"""The dbc package includes the modules for using external databases with Deephaven."""
+from deephaven import DHError
+from deephaven.table import Table
+import deephaven.arrow as dharrow
+
+
+def read_sql(conn: str, query: str) -> Table:
+    """Executes the provided SQL query via ConnectorX and returns a Deephaven table.
+
+    Args:
+        conn (str): a connection string URI, please refer to https://sfu-db.github.io/connector-x/databases.html for
+            database specific format
```

Comment:
This method is written in a very connectorx centric way, but it is a very flexible API.  For example, it could easily be made to work with ADBC and turbodbc.  I can see adding a third argument that indicates the connection type (and defaults to connectorx).  Then users have one powerful API that works with all of our connectors.  Is there a reason not to do this?

Reply:
It really isn't much additional complexity.

Reply:
I am in support of a simply interface for ODBC/ADBC but the flag pattern isn't so nice. How about read_odbc and read_adbc with the same signature as read_sql?  This will still enable us to check for the requirement of installation of turbodbc and adbc.